src/components/IOUConfirmationList.js

```diff
@@ -0,0 +1,248 @@
+                `$${this.calculateAmount() / 100}`);
+
+            sections.push({
+                title: 'WHO PAID?',
+                data: formattedMyPersonalDetails,
+                shouldShow: true,
+                indexOffset: 0,
+            });
+            sections.push({
+                title: 'WHO WAS THERE?',
+                data: formattedParticipants,
+                shouldShow: true,
+                indexOffset: 0,
+            });
+        } else {
+        // $ should be replaced by currency symbol once available
+            const formattedParticipants = getIOUConfirmationOptionsFromParticipants(this.props.participants,
+                `$${this.props.iouAmount}`);
+
+            sections.push({
+                title: 'TO',
+                data: formattedParticipants,
+                shouldShow: true,
+                indexOffset: 0,
+            });
+        }
+        return sections;
+    }
+
+    /**
+     * Gets splits for the transaction
+     *
+     * @returns {Array}
+     */
+    getSplits() {
+        const splits = this.props.participants.map(participant => ({
+            email: participant.login,
+
+            // we should send in cents to API
+            amount: this.calculateAmount(),
+        }));
+
+        splits.push({
+            email: this.props.myPersonalDetails.login,
+
+            // the user is default and we should send in cents to API
+            amount: this.calculateAmount(true),
+        });
+        return splits;
+    }
+
+    /**
+     * Gets participants list for a report
+     *
+     * @returns {Array}
+     */
+    getParticipants() {
+        const participants = this.props.participants.map(participant => participant.login);
+        participants.push(this.props.myPersonalDetails.login);
+        return participants;
+    }
+
+    /**
+     * Returns selected options with all participant logins
+     * @returns {Array}
+     */
+    getAllOptionsAsSelected() {
+        return [...this.props.participants,
+            getIOUConfirmationOptionsFromMyPersonalDetail(this.props.myPersonalDetails)];
+    }
+
+    /**
+     * Calculates the amount per user
+     * @param {Boolean} isDefaultUser
+     * @returns {Number}
+     */
+    calculateAmount(isDefaultUser = false) {
+        // convert to cents before working with iouAmount to avoid
+        // javascript subtraction with decimal problem
+        const iouAmount = Math.round(parseFloat(this.props.iouAmount * 100));
+        const totalParticipants = this.props.participants.length + 1;
+        const amountPerPerson = Math.round(iouAmount / totalParticipants);
+        const sumAmount = amountPerPerson * totalParticipants;
+        const difference = iouAmount - sumAmount;
+
+        if (!isDefaultUser) { return amountPerPerson; }
+
+        return iouAmount !== sumAmount ? (amountPerPerson + difference) : amountPerPerson;
+    }
+
+    render() {
+        const sections = this.getSections();
```

Comment:
Any reason to not just declare the prop like this: `sections={this.getSections()}`?